conftest.py

```diff
@@ -0,0 +1,193 @@
+"""
+import os
+import shutil
+import tempfile
+
+import pytest
+import yaml
+
+from yt.config import ytcfg
+from yt.utilities.answer_testing import utils
+
+
+# Global variables can be added to the pytest namespace
+pytest.answer_files = {}
+
+# List of answer files
+answer_file_list = 'tests/tests.yaml'
+answer_dir = os.path.join(ytcfg.get('yt', 'test_data_dir'), 'answers')
+
+
+#============================================
+#             pytest_addoption
+#============================================
+def pytest_addoption(parser):
+    """
+    Lets options be passed to test functions.
+    """
+    parser.addoption(
+        "--with-answer-testing",
+        action="store_true",
+        default=False,
+    )
+    parser.addoption(
+        "--answer-store",
+        action="store_true",
+        default=False,
+    )
+    parser.addoption(
+        "--answer-big-data",
+        action="store_true",
+        default=False,
+    )
+
+
+#============================================
+#             pytest_configure
+#============================================
+def pytest_configure(config):
+    """
+    Reads in the tests/tests.yaml file. This file contains a list of
+    each answer test's answer file (including the changeset number).
+    """
+    # Make sure that the answers dir exists. If not, try to make it
+    if not os.path.isdir(answer_dir):
+        os.mkdir(answer_dir)
+    # Read the list of answer test classes and their associated answer
+    # file
+    with open(answer_file_list, 'r') as f:
+        pytest.answer_files = yaml.safe_load(f)
+    # Register custom marks for answer tests and big data
+    config.addinivalue_line('markers', 'answer_test: Run the answer tests.')
+    config.addinivalue_line('markers', 'big_data: Run answer tests that require'
+        ' large data files.')
+
+
+#============================================
+#     pytest_collection_modifyitems
+#============================================
+def pytest_collection_modifyitems(config, items):
+    """
+    Decide which tests to skip based on command-line options.
+    """
+    # Set up the skip marks
+    skip_answer = pytest.mark.skip(reason="--with-answer-testing not set.")
+    skip_big = pytest.mark.skip(reason="--answer-big-data not set.")
+    # Loop over every collected test function
+    for item in items:
+        # If it's an answer test and the appropriate CL option hasn't
+        # been set, skip it
+        if "answer_test" in item.keywords and not config.getoption("--with-answer-testing"):
+            item.add_marker(skip_answer)
+        # If it's an answer test that requires big data and the CL
+        # option hasn't been set, skip it
+        if "big_data" in item.keywords and not config.getoption("--answer-big-data"):
+            item.add_marker(skip_big)
+
+
+#============================================
+#                   temp_dir
+#============================================
```

Comment:
Is there a good reason not to use std lib here ?
https://docs.python.org/3.7/library/tempfile.html

Reply:
I think it is, right?  It's using `tempfile.mkdtemp()` if not using specific arguments.

Reply:
oops, went much to fast here, sorry.